pkg/cmd/spks.go

```diff
@@ -55,142 +54,145 @@ func SpksCMD() *cli.Command {
 				EnvVars: []string{"UNIT_ID"}, Destination: &UnitID, Required: false, DefaultText: defaultTextForRequiredFlags, Value: "uom_uom_68_b1811ca1"},
 		},
 		Action: func(c *cli.Context) error {
+			ctxx, cancel := context.WithCancel(ctx)
+			defer cancel()
 			logger := log.Logger(c.Context)
 			logger.Info("starting spks data collector")
 
 			location, err := time.LoadLocation("Europe/Zurich")
 			if err != nil {
 				return fmt.Errorf("load location: %w", err)
 			}
-			// this variable is necessary to query Prometheus, with timerange [1d:1d] it returns data from 1 day up to midnight
-			startOfToday := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, location)
-			startYesterdayAbsolute := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day()-1, 0, 0, 0, 0, location).In(time.UTC)
-			endYesterdayAbsolute := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day()-1, 23, 59, 59, 0, location).In(time.UTC)
-
-			logger.Info("Getting specific metric from thanos")
-
-			client, err := api.NewClient(api.Config{
-				Address: prometheusURL,
-			})
-			if err != nil {
-				logger.Error(err, "Error creating Prometheus client")
-				return err
-			}
 
-			v1api := v1.NewAPI(client)
-			ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
-			defer cancel()
-			// send this value to odoo
-
-			mariadbStandard, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[0], logger, startOfToday)
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
-			mariadbPremium, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[1], logger, startOfToday)
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
-			redisStandard, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[2], logger, startOfToday)
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
-			redisPremium, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[3], logger, startOfToday)
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
+			now := time.Now().In(location)
```

Comment:
Did you check with tobru if the timestamps are now really correct?

Reply:
We had to change a bit of code on Monday because the timestamp was showing 00:00Z instead of 22:00Z UTC. Now data is arriving properly.

Reply:
Yes it's correct as it is now. We tested and verified it in Odoo directly.